underfs/swift/src/main/java/alluxio/underfs/swift/SwiftInputStream.java

```diff
@@ -92,33 +99,44 @@ public long skip(long n) throws IOException {
     if (n <= 0) {
       return 0;
     }
+
+    LOG.debug("Swift InputStream {}: skipping {} bytes", this.hashCode(), n);
     closeStream();
     mPos += n;
     openStream();
+    LOG.debug("Swift InputStream {}: done skipping", this.hashCode());
     return n;
   }
 
   /**
    * Opens a new stream at mPos if the wrapped stream mIn is null.
    */
   private void openStream() {
+    LOG.debug("Swift InputStream {}: open stream at pos {}", this.hashCode(), mPos);
+
     if (mStream != null) { // stream is already open
+      LOG.debug("Swift InputStream {}: stream is already open", this.hashCode());
       return;
     }
     StoredObject storedObject = mAccount.getContainer(mContainerName).getObject(mObjectPath);
     DownloadInstructions downloadInstructions  = new DownloadInstructions();
-    downloadInstructions.setRange(new ExcludeStartRange(mPos));
+    final long endPos = mPos + Configuration.getBytes(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT);
```

Comment:
you're right that it will fail to read if user tries to read more than provisioned. but when does that happen?. will alluxio not try to read from UFS in chunks of blocksize. 

although it is still true that opening a block for read and closing it without reading entire block is still inefficient.